server/src/main/java/com/tencent/rss/server/ShuffleServerConf.java

```diff
@@ -258,6 +258,44 @@
       .defaultValue(2 * 1024L * 1024L)
       .withDescription("The index file size hint");
 
+  public static final ConfigOption<Double> RSS_HEALTH_STORAGE_MAX_USAGE_PERCENTAGE = ConfigOptions
+      .key("rss.server.health.max.storage.usage.percentage")
+      .doubleType()
+      .checkValue(ConfigUtils.percentageDoubleValidator,
+          "The max usage percentage must be between 0.0 and 100.0")
+      .defaultValue(90.0)
+      .withDescription("The usage percentage of a storage exceed the value, the disk become unavailable");
+
+  public static final ConfigOption<Double> RSS_HEALTH_STORAGE_RECOVERY_USAGE_PERCENTAGE = ConfigOptions
+      .key("rss.server.health.storage.recovery.usage.percentage")
+      .doubleType()
+      .checkValue(ConfigUtils.percentageDoubleValidator,
+          "The recovery usage percentage must be between 0.0 and 100.0")
+      .defaultValue(80.0)
+      .withDescription("The usage percentage of an unavailable storage decline the value, the disk"
+          + " will become available");
+
+  public static final ConfigOption<Long> RSS_HEALTH_CHECK_INTERVAL = ConfigOptions
+      .key("rss.server.health.check.interval.ms")
+      .longType()
+      .checkValue(ConfigUtils.positiveLongValidator,  "The interval for health check must be positive")
+      .defaultValue(5000L)
+      .withDescription("The interval for health check");
+
+  public static final ConfigOption<Double> RSS_HEALTH_MIN_STORAGE_PERCENTAGE = ConfigOptions
+      .key("rss.server.health.min.storage.percentage")
+      .doubleType()
+      .checkValue(ConfigUtils.percentageDoubleValidator,
+          "The minimum for healthy storage percentage must be between 0.0 and 100.0")
+      .defaultValue(80.0)
+      .withDescription("The minimum fraction of storage that must pass the check mark the node as healthy");
+
+  public static final ConfigOption<Boolean> RSS_USE_HEALTH_CHECK = ConfigOptions
```

Comment:
This feature should bind with local storage, always enable if local storage is enabled

Reply:
Both LocalStorage and  MultiStorage need this feature. And I check storage type when shuffle server start.